src/libraries/Common/src/System/Net/Security/Unix/SafeFreeCredentials.cs

```diff
@@ -18,8 +19,13 @@ internal abstract class SafeFreeCredentials : DebugSafeHandle
     internal abstract class SafeFreeCredentials : SafeHandle
     {
 #endif
+        internal DateTime _expiry;
```

Comment:
I know this is tricky but AFAIK Unix does not have the same issue as Windows. I'm wondering if we should (need to?) replicate the logic.

Reply:
I wasn't sure how to do this so that the code compiles for both platforms, since the code for `SslSessionsCache` where I access the Expiry property is not platform specific. However, it would be possible to always leave MaxValue on non-windows platforms and set the expiry in SslStreamPal.Windows.cs